WD-Firmware/comrv/comrv.c

```diff
@@ -18,34 +18,884 @@
+   comrvCacheEntry_t *pCacheEntry;
+
+   /* loop all entries */
+   for (ucEntryIndex = 0 ; ucEntryIndex < D_COMRV_NUM_OF_CACHE_ENTRIES ; ucEntryIndex+=pCacheEntry->unProperties.stFields.ucSizeInMinGroupSizeUnits)
+   {
+      pCacheEntry = &g_stComrvCB.stOverlayCache[ucEntryIndex];
+      /* if token already loaded */
+      if (pCacheEntry->unToken.stFields.overlayGroupID == unToken.stFields.overlayGroupID)
+      {
+         /* return the actual function location within the loaded overlay group */
+         return ucEntryIndex;
+      }
+   }
+
+   /* overlay group not loaded */
+   return D_COMRV_GROUP_NOT_FOUND;
+}
+
+/**
+* Update a given comrv cache entry was accessed
+*
+* @param entryIndex - the comrv cache entry being accessed
+*
+* @return none
+*/
+static void comrvUpdateCacheEvectionParams(u08_t ucEntryIndex)
+{
+   comrvCacheEntry_t *pCacheEntry;
+
+#ifdef D_COMRV_EVICTION_LRU
+
+   /* there is no need to update if ucEntryIndex is already MRU */
+   if (ucEntryIndex !=  g_stComrvCB.ucMruIndex)
+   {
+      pCacheEntry = &g_stComrvCB.stOverlayCache[ucEntryIndex];
+      /* ucEntryIndex is not the ucLruIndex */
+      if (ucEntryIndex !=  g_stComrvCB.ucLruIndex)
+      {
+         /* update previous item's 'next index' */
+         g_stComrvCB.stOverlayCache[pCacheEntry->unLru.stFields.typPrevLruIndex].unLru.stFields.typNextLruIndex =
+               pCacheEntry->unLru.stFields.typNextLruIndex;
+      }
+      else
+      {
+         /* update the global lru index */
+         g_stComrvCB.ucLruIndex = pCacheEntry->unLru.stFields.typNextLruIndex;
+         /* update the lru item with the previous item index */
+         g_stComrvCB.stOverlayCache[g_stComrvCB.ucLruIndex].unLru.stFields.typPrevLruIndex = D_COMRV_LRU_ITEM;
+      }
+      /* update ucEntryIndex previous index */
+      pCacheEntry->unLru.stFields.typPrevLruIndex = g_stComrvCB.ucMruIndex;
+      /* update ucEntryIndex next index - last item (MRU)*/
+      pCacheEntry->unLru.stFields.typNextLruIndex = D_COMRV_MRU_ITEM;
+      /* update the old mru's next index */
+      g_stComrvCB.stOverlayCache[g_stComrvCB.ucMruIndex].unLru.stFields.typNextLruIndex = ucEntryIndex;
+      /* update the new MRU */
+      g_stComrvCB.ucMruIndex = ucEntryIndex;
+   }
+
+#elif defined(D_COMRV_EVICTION_LFU)
+#elif defined(D_COMRV_EVICTION_MIX_LRU_LFU)
+#endif /* D_COMRV_EVICTION_LRU */
+}
+
+/**
+* get comrv status
+*
+* @param none
+*
+* @return none
+*/
+void comrvGetStatus(void)
+{
+   // TODO: check comrv stack
+}
+
+/**
+* initialize comrv stack - needs to be invoke by each task (if rtos exist)
+* when before initializing task stack.
+* in os-less apps, this function is called by comrv initialization function
+* and the user application doesn't need to do that.
+*
+* @param none
+*
+* @return none
+*/
+D_COMRV_NO_INLINE void comrvInitApplicationStack(void)
+{
+   comrvStackFrame_t *pStackPool, *pStackFrame;
+
+   /* disable ints */
+   // TODO: disable ints
```

Comment:
add "todo", with rtosal, implement condition check if scheduler is running or not, if running use sync api

Reply:
done in ca8dbd0